kroxylicious-docs/docs/_modules/record-encryption/azure-key-vault/con-azure-key-vault-key-creation.adoc

```diff
@@ -0,0 +1,51 @@
+:_mod-docs-content-type: CONCEPT
+
+// file included in the following:
+//
+// assembly-azure-key-vault.adoc
+
+[id='con-azure-key-vault-key-creation-{context}']
+= Creating Azure Key Vault keys
+
+[role="_abstract"]
+As a user with the *Key Vault Crypto Officer* role (or otherwise sufficient {microsoft-learn}/azure/key-vault/general/rbac-guide?tabs=azure-cli#azure-built-in-roles-for-key-vault-data-plane-operations[RBAC] or {microsoft-learn}/azure/key-vault/general/rbac-access-policy[legacy] permissions to manage keys in Azure Key Vault), use either the {microsoft-learn}/cli/azure/keyvault/key?view=azure-cli-latest#az-keyvault-key-create[Azure CLI] or {microsoft-learn}/powershell/module/az.keyvault/add-azkeyvaultkey[Azure PowerShell] to create a *key* with *wrapKey* and *unwrapKey* operations enabled.
+
+== Establish the naming convention for keys within key vault hierarchy
+
+Establish a naming convention for keys to keep the filter’s keys separate from those used by other systems.
+
+NOTE: The filter key name prefix `KEK_` is used widely across the Kroxylicious project, including here in the documentation, as well as in the quickstarts, developer guides, examples, and scripts. However, *Azure Key Vault does not permit underscore characters in key names*, so Azure Key Vault users will have to adjust this naming scheme as required.
+
+== Supported Key Types
+
+NOTE: The key types available to you in Azure Key Vault will depend on your Azure Key Vault service tier. See the {microsoft-learn}/azure/key-vault/keys/about-keys[_About keys_ page in the Azure Key Vault docs] for more details.
+
+The filter will accept the following {microsoft-learn}/rest/api/keyvault/keys/create-key/create-key?view=rest-keyvault-keys-2025-07-01&tabs=HTTP#jsonwebkeytype[key types] (along with their associated {microsoft-learn}/rest/api/keyvault/keys/wrap-key/wrap-key?view=rest-keyvault-keys-2025-07-01&tabs=HTTP#jsonwebkeyencryptionalgorithm[wrapping algorithms]) from Azure Key Vault:
+
+* *Symmetric Keys (`oct` or `oct-HSM`):* Uses the quantum-resistant `A256GCM` (256-bit AES GCM) wrapping algorithm, which {microsoft-learn}/azure/key-vault/keys/about-keys-details#symmetric-key-algorithms-managed-hsm-only[requires a Managed HSM subscription].
+* *Asymmetric Keys (`RSA` or `RSA-HSM`):* Uses the `RSA-OAEP-256` (RSAES-OAEP with SHA-256 hash and MGF1/SHA-256 mask) wrapping algorithm, which is not quantum-resistant. This wrapping algorithm {microsoft-learn}/azure/key-vault/keys/about-keys-details#wrapkeyunwrapkey-encryptdecrypt[does not require a Managed HSM subscription to use] (however, `RSA-HSM` is not available without a premium Azure Key Vault subscription).
+
+If using the Azure CLI, creating a key can be accomplished with a command like this:
+
+[source,shell]
+----
+KEY_VAULT_NAME="kroxylicious-key-vault"
+KEY_NAME="KEK_<name>"
+KEY_TYPE="oct" # <1>
```

Comment:
I wonder if we should change the example to `rsa` despite it being non quantum resistant. I don't imagine there's much chance of a user trying this out for the first time on an HSM instance and using `oct` throws up an exception with the keyvault we created earlier.